subtrack.MAUI/Services/SubscriptionsCalculator.cs

```diff
@@ -51,14 +51,10 @@ public decimal GetYearlyCost(Subscription subscription)
 
     public IEnumerable<Subscription> GetSubscriptionListByMonth(IEnumerable<Subscription> subscriptions, DateTime monthDate)
     {
-        bool paidThisMonth(Subscription s) => s.LastPayment.Month == monthDate.Month && s.LastPayment.Year == monthDate.Year;
-        bool paymentStartsInTheFuture(Subscription s) => s.LastPayment > monthDate;
-
-        var subscriptionsByMonth = subscriptions
-                    .Where(s => !paidThisMonth(s) && !paymentStartsInTheFuture(s))
-                    .ToList();
-
-        return subscriptionsByMonth;
+        return subscriptions
+                    .Where(subscription => PaidInPast(subscription, monthDate) || (subscription.BillingOccurrence == BillingOccurrence.Week && PaidThisMonth(subscription, monthDate)))
```

Comment:
This logic is a bit broken, the previously mentioned sub now displays its last payment on the monthly page. Im gonna try to see how it can all be written in a more readable manner